### PR TITLE
Change: Option to only show Ender Node tracker while holding a pickaxe

### PR DESCRIPTION
**Pull request by Thunderblade73**
## Discord
https://discord.com/channels/997079228510117908/1197526187157758033

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/config/features/combat/EnderNodeConfig.java [hunk 1]
```diff
@@ -41,6 +41,14 @@ public class EnderNodeConfig {
     @FeatureToggle
     public boolean enabled = false;
 
+    @Expose
+    @ConfigOption(
+        name = "Only Holding Tool",
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/combat/EnderNodeConfig.java]: Maybe this could be only while holding tool

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
Just need to change this feature name to be a bit more descriptive

File: src/main/java/at/hannibal2/skyhanni/features/combat/endernodetracker/EnderNodeTracker.kt [hunk 2]
```diff
@@ -184,7 +183,9 @@ object EnderNodeTracker {
         return newProfit
     }
 
-    private fun isInTheEnd() = LorenzUtils.skyBlockArea == "The End"
+    private fun isEnabled() =
+        LorenzUtils.inSkyBlock && IslandType.THE_END.isInIsland() && config.enabled &&
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/combat/endernodetracker/EnderNodeTracker.kt]: .isInIsland already checks for being in skyblock

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
also pushed one punctuation change (before i saw the other thing needed changing)

File: src/main/java/at/hannibal2/skyhanni/config/features/combat/EnderNodeConfig.java [hunk 3]
```diff
@@ -41,6 +41,14 @@ public class EnderNodeConfig {
     @FeatureToggle
     public boolean enabled = false;
 
+    @Expose
+    @ConfigOption(
+        name = "Only While Holding Tool",
+        desc = "Only shows the tracker if holding a pickaxe, drill or gauntlet."
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/config/features/combat/EnderNodeConfig.java]: Maybe add "in hand" at the end

File: src/main/java/at/hannibal2/skyhanni/features/combat/endernodetracker/EnderNodeTracker.kt [hunk 4]
```diff
@@ -184,7 +183,9 @@ object EnderNodeTracker {
...
+            (!config.onlyPickaxe || ItemCategory.miningTools.containsItem(InventoryUtils.getItemInHand()))
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/combat/endernodetracker/EnderNodeTracker.kt]: Maybe extract into function "hasItemInHand()" for readability

File: src/main/java/at/hannibal2/skyhanni/utils/ItemCategory.kt [hunk 5]
```diff
@@ -48,4 +51,12 @@ enum class ItemCategory {
 
     NONE,
     ;
+
+    companion object {
+
+        fun Collection<ItemCategory>.containsItem(stack: ItemStack?) =
+            stack?.getItemCategoryOrNull()?.let { this.contains(it) } ?: false
+
+        val miningTools = listOf(ItemCategory.PICKAXE, ItemCategory.DRILL, ItemCategory.GAUNTLET)
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/ItemCategory.kt]: Remove redundant qualifier name

**Review by hannibal002 — CHANGES_REQUESTED** [review]
It is working fine. Only small cleanup ideas.

**Review by CalMWolfs — APPROVED** [review]
lgtm

File: src/main/java/at/hannibal2/skyhanni/features/combat/endernodetracker/EnderNodeTracker.kt (hunk 4 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/combat/endernodetracker/EnderNodeTracker.kt]: I was thinking of this, but It's fine.
image

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**